test/unit_test/data/repository/weather_repository_test.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_training/data/app_exception.dart';
+import 'package:flutter_training/data/data_source/weather_data_source.dart';
+import 'package:flutter_training/data/models/app_api_result.dart';
+import 'package:flutter_training/data/models/weather/weather.dart';
+import 'package:flutter_training/data/models/weather/weather_request.dart';
+import 'package:flutter_training/data/repository/weather_repository.dart';
+import 'package:flutter_training/views/constants/strings.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'weather_repository_test.mocks.dart';
+
+@GenerateMocks([WeatherDataSource])
+void main() {
+  final defaultRequest = WeatherRequest(
+    date: DateTime.now(),
+  );
+  final client = MockWeatherDataSource();
```

Comment:
[nits]
名前的に dataSource があっていそうです！

[ask]
あと、ここで Repository のインスタンスも生成していたほうがいいかなと思いますが、いかがでしょうか？

```suggestion
  final dataSource = MockWeatherDataSource();
  final repository = WeatherRepository(dataSource);
```

Reply:
修正いたしました。
[fetchWeatherUseCase](https://github.com/trm11tkr/flutter-training-yumemi/blob/session/9/test/unit_test/data/use_case/fetch_weather_use_case_test.dart)のclientも同様に修正いたしました！